File: SelectionSort.java
public class SelectionSort {
    static void swap(int[] arr, int a, int b) {
        int temp = arr[a];
        arr[a] = arr[b];
        arr[b] = temp;
    }

    static void selectionSort(int[] arr) {
        int size = arr.length;
        for (int i = 0; i < size - 1; i++) {
            int minIndex = i;
            for (int j = i + 1; j < size; j++) {
                if (arr[j] < arr[minIndex]) {
                    minIndex = j;
                }
            }
            swap(arr, i, minIndex);
        }
    }

    static void displayArray(int[] arr) {
        for (int num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();
    }

    public static void main(String[] args) {
        int[] arr = { 45, 34, 23, 12, 76 };

        System.out.println("Original array:");
        displayArray(arr);

        selectionSort(arr);

        System.out.println("Sorted array:");
        displayArray(arr);
    }
}

File: SelectionSort.py
def swap(a, b):
    temp = a
    a = b
    b = temp


def selection_sort(arr):
    size = len(arr)
    for i in range(size - 1):
        min_index = i
        for j in range(i + 1, size):
            if arr[j] < arr[min_index]:
                min_index = j
        arr[i], arr[min_index] = arr[min_index], arr[i]


def display_array(arr):
    for num in arr:
        print(num, end=" ")
    print()


if __name__ == "__main__":
    arr = [45, 34, 23, 12, 76]

    print("Original array:")
    display_array(arr)

    selection_sort(arr)

    print("Sorted array:")
    display_array(arr)
